src/sdk/namespace_cache.js

```diff
@@ -4,19 +4,35 @@
 const _ = require('lodash');
 const stream = require('stream');
 const assert = require('assert');
+const dbg = require('../util/debug_module')(__filename);
 
 class NamespaceCache {
 
-    constructor({ namespace_hub, namespace_nb, active_triggers }) {
+    constructor({ namespace_hub, namespace_nb, caching, active_triggers }) {
         this.namespace_hub = namespace_hub;
         this.namespace_nb = namespace_nb;
         this.active_triggers = active_triggers;
+        this.caching = caching;
     }
 
     get_write_resource() {
         return this.namespace_hub;
     }
 
+    _delete_object_from_cache(params, object_sdk) {
+        process.nextTick(() => {
+            const delete_params = _.pick(params, 'bucket', 'key');
+            dbg.log0("NamespaceCache._delete_object_from_cache", delete_params);
+            this.namespace_nb.delete_object(delete_params, object_sdk)
+            .then(() => {
+                dbg.log0('NamespaceCache: deleted object from cache', delete_params);
+            })
+            .catch(err_delete => {
+                dbg.warn('NamespaceCache: error in deleting object from cache', params, err_delete);
+            });
```

Comment:
It depends on whether we would like to have s3 operation blocked by cache cleanup. See my response in https://github.com/noobaa/noobaa-core/pull/6004#discussion_r447018283

Reply:
Actually it doesn't really depend - you can still make the callback function invoked by nextTick to be async, clearly nextTick will not await for it, but it will allow your function to be written with await and try-catch inside which makes it easier to read

Reply:
Will change it to use try and catch.
Will NOT use await to call _delete_object_from_cache() in relevant places.